bart/sched/pelt.py

```diff
@@ -0,0 +1,347 @@
+    (e.g. half-life, min/max stability values)
+    """
+
+    _sample_us = 1024
+    _signal_max = 1024
+
+    def __init__(self, init_util=0, half_life_ms=32):
+        """
+        Initialize the PELT simulator for a specified task.
+
+        :param init_util: the initial utilization for the task
+        :type  init_util: int
+
+        :param half_life_ms: the [ms] interval required by a signal to
+                             increased/decreased by half of its range
+        :type  half_life_ms: int
+        """
+        self.init_util = init_util
+        self.half_life_ms = half_life_ms
+
+        self._geom_y = pow(0.5, 1./half_life_ms)
+        self._geom_u = float(Simulator._signal_max) * (1. - self._geom_y)
+
+        print "PELT Configured with   :"
+        print "  initial utilization  : ", self.init_util
+        print "  half life (HL)  [ms] : ", self.half_life_ms
+        print "  y =    0.5^(1/HL)    : {:6.3f}".format(self._geom_y)
+        print "  u = {:5d}*(1 - y)    : {:6.3f}".format(Simulator._signal_max, self._geom_u)
+
+        self._df = None
+
+    def _geom_sum(self, u_1, active_us=0):
+        """
+        Geometric add the specified value to the u_1 series.
+
+        NOTE: the current implementation assume that the task was active
+              (active_us != 0) or sleeping (active_us == 0) a full PELT
+              sampling interval (by default 1024[us])
+        """
+        # Decay previous samples
+        u_1 *= self._geom_y
+        # Add current sample (if task was active)
+        if active_us:
+            u_1 += self._geom_u
+        return u_1
+
+    def stable_range(self, task):
+        """
+        Compute the stability ranges for the specified (periodic) task
+
+        Here we use:
+        - Decay factor:       y = 0.5^(1/half_life)
+        - Max stable value:       (1-y^r) / (1-y^p)
+        - Min stable value: y^i * (1-y^r) / (1-y^p)
+
+        :param task: the task we want to simulate the PELT signal for, by
+                     default is the task used to intialize the Simulator
+                     (when specified)
+        :type  task: Task
+
+        :return: A tuple representing the minimum and maximum value of the PELT
+                 singal once stable.
+
+        """
+
+        # Validate input parameters
+        if not isinstance(task, Task):
+            raise ValueError("Wrong time for task parameter")
+
+        # Compute max value
+        max_util  = (1. - pow(self._geom_y, float(task.run_us)/Simulator._signal_max))
+        max_util /= (1. - pow(self._geom_y, float(task.period_us)/Simulator._signal_max))
+
+        # Compute min value, by decaying the maximum value
+        min_util  = (max_util * pow(self._geom_y, float(task.idle_us)/Simulator._signal_max))
+
+        return PELT_Range(Simulator._signal_max * min_util,
+                         Simulator._signal_max * max_util)
+
+
+    def signal(self, task, start_s=0, end_s=10):
```

Comment:
signal can be a verb as well. I think in this case we do need a get

so `getSignal`